File: catkin_ws/src/learning_launch/scripts/pose_subscriber.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# The routine will subscribe to/turtle1/pose topic, message type is sturtlesim::Pose

import rospy
from turtlesim.msg import Pose

def poseCallback(msg):
  rospy.loginfo("Turtle pose: x:%0.6f, y:%0.6f", msg.x, msg.y)

def pose_subscriber():
  # ROS init
  rospy.init_node('pose_subscriber', anonymous=True)

  # Create a Subscriber, subscribe to the topic /turtle1/pose, and register the callback function poseCallback
  rospy.Subscriber("/turtle1/pose", Pose, poseCallback)

  # Enter a loop, pumping callbacks
  rospy.spin()

if __name__ == '__main__':
  pose_subscriber()

File: catkin_ws/src/learning_topic/src/pose_subscriber.cpp
/**
 * The routine will subscribe to/turtle1/pose topic, message type is turtlesim::Pose
 */
 
#include <ros/ros.h>
#include "turtlesim/Pose.h"


//After receiving the subscribed message,
// enter the message callback function
void poseCallback(const turtlesim::Pose::ConstPtr& msg)
{
    // Print out the received message
    ROS_INFO("Turtle pose: x:%0.6f, y:%0.6f", msg->x, msg->y);
}

int main(int argc, char **argv)
{
    // ROS init
    ros::init(argc, argv, "pose_subscriber");

    // Create a node handle
    ros::NodeHandle n;

    // Create a Subscriber, subscribe to the topic /turtle1/pose,
    // and register the callback function poseCallback
    ros::Subscriber pose_sub = n.subscribe("/turtle1/pose", 10, poseCallback);

    // Enter a loop, pumping callbacks
    ros::spin();

    return 0;
}
